File: 53. 最大子数组和/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # 局部最优：当前“连续和”为负数的时候立刻放弃，从下一个元素重新计算“连续和”，因为负数加上下一个元素 “连续和”只会越来越小。
        ans = float('-inf')
        n = len(nums)
        if n == 1:
            return nums[0]

        Sum = 0
        for i in range(n):
            Sum += nums[i]
            ans = max(ans, Sum)
            if Sum < 0:
                Sum = 0
        return ans

File: 53. 最大子数组和/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans = Integer.MIN_VALUE;
        int n = nums.length;
        if(n == 1) return nums[0];
        int Sum = 0;
        for(int i = 0; i < n; i++){
            Sum += nums[i];
            ans = Math.max(ans, Sum);
            if(Sum < 0) Sum = 0;
        }
        return ans;
    }
}